tests/PerformanceTests/Microsoft.Identity.Web.Perf.Client/ScalableTokenCacheHelper.cs

```diff
@@ -38,15 +38,25 @@ static class ScalableTokenCacheHelper
         private static ConcurrentDictionary<string, string> s_tokenCacheKeys = new ConcurrentDictionary<string, string>();
         private static string s_emptyContent = " ";
 
-        private static volatile bool s_isPersisting = false; 
+        private static volatile bool s_isPersisting = false;
+        private static object s_persistLock = new object();
 
         internal static void PersistCache()
```

Comment:
This smells of double check locking, which is pretty difficult to get right. Using Lazy<T> is an easier way to achieve all this in a thread-safe manner. 

See https://www.sudhanshutheone.com/posts/double-check-lock-csharp

Reply:
It's not just an initializer problem Bogdan. This code is run continuously

---
In reply to: [510004073](https://github.com/AzureAD/microsoft-identity-web/pull/710#discussion_r510004073) [](ancestors = 510004073)